pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/PendingAckHandleImpl.java

```diff
@@ -181,7 +181,7 @@ private void addIndividualAcknowledgeMessageRequest(TxnID txnID,
     public void internalIndividualAcknowledgeMessage(TxnID txnID, List<MutablePair<PositionImpl, Integer>> positions,
                                                      CompletableFuture<Void> completableFuture) {
         if (txnID == null) {
-            completableFuture.completeExceptionally(new NotAllowedException("Positions can not be null."));
+            completableFuture.completeExceptionally(new NotAllowedException("txnID can not be null."));
```

Comment:
Could you explain it in more detail? Why changed here?

Reply:
The condition checks if txnID is null, not the Positions, isn't it?